### ISSUE TITLE
BUG - Expand on draft entity returns active entity for 1:1 compositions

### ISSUE THREAD

**Issue by nilsbrenker** [SAP/open-ux-odata#221]
### Description
Selecting the inactive entity of a draft enabled Odata v4 entity with an expand to a 1:1 composition (`childEntity`) returns the active version of the `childEntity` instead of the inactive one. Expands for 1:n compositions are working as expected and I can also see that the inactive entity got created, just the linking seems to be incorrect.

### Expected results

CDS model:
```
entity RootEntity: cuid, managed {
  childEntity: Composition of one ChildEntity;  
  childEntity2: Composition of many ChildEntity2 on childEntity2.root = $self;
}
```

`GET RootEntity(ID=75a043dc-beae-4d64-9897-ae773fa006a6,IsActiveEntity=false)?$expand=childEntity`

```
{
  ID: "75a043dc-beae-4d64-9897-ae773fa006a6",
  IsActiveEntity: false,
  HasActiveEntity: true,
  HasDraftEntity: true,
  childEntity: {
    ID: "b4a3a462-3131-4b05-a79b-6321e04ec076",
    IsActiveEntity: false,
    HasActiveEntity: true,
    HasDraftEntity: true
  }
}
```

### Actual results

```
{
  ID: "75a043dc-beae-4d64-9897-ae773fa006a6",
  IsActiveEntity: false,
  HasActiveEntity: true,
  HasDraftEntity: true,
  childEntity: {
    ID: "b4a3a462-3131-4b05-a79b-6321e04ec076",
    IsActiveEntity: true,
    HasActiveEntity: true,
    HasDraftEntity: true
  }
}
```

But a `GET ChildEntity` returns both the active and inactive entity, so the draft is existing.

#### Version/Components/Environment	
Running on latest mockserver version 2.1.22

### Problem
{describe the problem} 

### Fix
{describe the fix} 

### Why was it missed
{Some explanation why this issue might have been missed during normal development/testing cycle}  

### How can we avoid this
{if we don’t want to see this type of issues anymore what we should do to prevent}

**Comment by nlunets** [comment]
Hi @nilsbrenker this should be fixed with change #223 :)

**Comment by nlunets** [comment]
Fixed in version 2.1.23

**Comment by nilsbrenker** [comment]
I can confirm that the issue is solved, thank you @nlunets :)
